lib/graphql/graphql-client.ts

```diff
@@ -0,0 +1,46 @@
+import axios, { AxiosRequestConfig, AxiosResponse } from 'axios'
+
+import { GraphQLResponse } from './graphql-schemas'
+
+/* Interface for accessing any GraphQL API */
+export interface IGraphQLClient {
+  fetchData<T>(query: string, variables?: { [key: string]: any }): Promise<T>
+}
```

Comment:
In SOR we use this library for graphql requests: https://github.com/jasonkuhrt/graphql-request

I really have no strong opinion either way, but I just wanted to throw it out there too in case you were not aware :)

Reply:
thanks - wasn't aware. Will keep as is for now as it's a simple implementation + tested.
might refactor in the future though if we need any extra functionality

Reply:
I think routing-api is better off just using the axios library for remote calls, unless we see a strong benefit of DevEx in using a particular library like typechain for ease of RPC calls compositions.

otoh, if in future, we have many GQL upstream dependencies, we can push the data team to implement a custom GQL client for every upstream to consume, including us. Until then, I'm in favor of less external library dependencies in routing-api/SOR, especially we already have many library dependencies.